src/parser.rs

```diff
@@ -2562,10 +2568,38 @@ impl<'a> Parser<'a> {
             None
         };
         match self.next_token() {
+            Token::Word(w)
+                if (w.keyword == Keyword::KEY || w.keyword == Keyword::INDEX)
+                    && dialect_of!(self is MySqlDialect) =>
+            {
+                let name = Some(self.parse_identifier()?);
+                let columns = self.parse_parenthesized_column_list(Mandatory)?;
+                match w.keyword {
+                    Keyword::KEY => Ok(Some(TableConstraint::Key {
+                        name,
+                        columns,
+                        format: KeyFormat::Key,
+                    })),
+                    Keyword::INDEX => Ok(Some(TableConstraint::Key {
+                        name,
+                        columns,
+                        format: KeyFormat::Index,
+                    })),
+                    _ => unreachable!(),
```

Comment:
Please replace this with an error. I left a possible approach for cases like that [here](https://github.com/sqlparser-rs/sqlparser-rs/pull/608#discussion_r985164256)

Reply:
There would be only two cases, and both of it are covered, so it's actually unreachable.

Reply:
Fine, but can't you return an error here?

It's better to have an error (like the `self.expected(...)`) than having a unreachable